airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/thread_based_concurrent_stream.py

```diff
@@ -105,9 +109,11 @@ def read(self) -> Iterable[Record]:
                         f"Received sentinel for partition {record_or_partition.partition} that was not in partitions. This is indicative of a bug in the CDK. Please contact support.partitions:\n{partitions_to_done}"
                     )
                 partitions_to_done[record_or_partition.partition] = True
+                self._cursor.close_partition(record_or_partition.partition)
```

Comment:
This leads to another discrepancy where the state message is emitted after the `{"type": "TRACE", "trace": {"type": "STREAM_STATUS", "emitted_at": x, "stream_status": {"stream_descriptor": {"name": <stream_name>, "namespace": null}, "status": "COMPLETE"}}}`

I don't think this would be a problem but I wanted to raise awareness on that just in case

Reply:
good to know, but I agree that this is not a problem.